src/language-js/print/function.js

```diff
@@ -176,16 +178,24 @@ function printArrowFunctionSignature(path, options, print, args) {
   if (shouldPrintParamsWithoutParens(path, options)) {
     parts.push(print(["params", 0]));
   } else {
+    const expandArg = args && (args.expandLastArg || args.expandFirstArg);
+    let returnTypeDoc = printReturnType(path, print, options);
+    if (expandArg) {
+      if (willBreak(returnTypeDoc)) {
+        throw new ArgExpansionBailout();
+      }
```

Comment:
Can you add tests that matches this condition?

Reply:
done